### PR TITLE
Update 'SDL2' package to v2.0.18

### PR DESCRIPTION
**Pull request by drodin**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/update.html)
  step by step carefully. **[Yes]**

From official repo.

---

### CODE REVIEW

File: cmake/projects/SDL2/hunter.cmake [hunk 1]
```diff
@@ -149,6 +149,17 @@ hunter_add_version(
         3b8d0dfdbd1b062035a9432bff75a22ab439bcd1
 )
 
+hunter_add_version(
+        PACKAGE_NAME
+        SDL2
+        VERSION
+        "2.0.18"
+        URL
+        "https://github.com/libsdl-org/SDL/archive/release-2.0.18.tar.gz"
```
Comment (NeroBurner) [inline cmake/projects/SDL2/hunter.cmake:158]: SDL can now be used directly from upstream? I doubt SDL has no external dependencies for building. Or did they themself add Hunter support?

Comment (drodin) [reply cmake/projects/SDL2/hunter.cmake:158]: Yes. It has no external dependencies. 2.0.12 only needed an alias for static lib (SDL2::SDL2 to SDL2::SDL2-static, for hunter backward compatibility), same as 2.0.16 (2.0.14 cmake was broken a lot, so I didn't bother). They finally put the exact same alias in SDL2Config.cmake in 2.0.18, so at least this version can be used from upstream.

Comment (NeroBurner) [reply cmake/projects/SDL2/hunter.cmake:158]: That's awesome! Thank you!

**Review by NeroBurner — APPROVED** [review]

**Review by rbsheth — APPROVED** [review]

**Comment by rbsheth** [comment]
Released: https://github.com/cpp-pm/hunter/releases/tag/v0.23.320

**Pull request merged**